Dockerfile.c12e-ci.cortex-certifai-reference-model-server.ubi8

```diff
@@ -1,53 +1,55 @@
-FROM registry.access.redhat.com/ubi9/ubi-minimal:latest
+####################################################
+## Build container
+####################################################
 
-LABEL release=1 \
-  name="cortex-certifai-reference-models-server" \
-  vendor="CognitiveScale" \
-  version=1 \
-  summary="Cortex Certifai Reference Model Server" \
-  description="Reference Model Server for Cognitive Scale's Cortex Certifai" \
-  com.cognitivescale.license_terms="https://www.cognitivescale.com/legal-information/"
+FROM c12e/certifai:base-build AS build
 
-COPY licenses /licenses
 COPY artifacts/packages/cortex-certifai-reference-model-server-*.zip \
        /tmp/
 
-WORKDIR /app
-
-USER root
-
-RUN microdnf update -y && microdnf install wget findutils -y
+RUN conda run -n certifai pip install --no-cache-dir --upgrade pip
+RUN conda run -n certifai pip install --no-cache-dir \
+     $(find /tmp -name cortex-certifai-reference-model-server-*.zip)
 
-# Create '/conda' folder '.bashrc' for the non-root user, otherwise they cannot use conda
-RUN mkdir /conda && chown -R 1001 /conda && chmod -R u+w /conda && \
-    touch //.bashrc && chown -R 1001 //.bashrc && chmod u+w //.bashrc
+# Use conda-pack to save the conda env in a standalone environment (/venv).
+# NOTE: The environment must be in the same path in the runtime container.
+RUN conda-pack -n certifai --ignore-missing-files -o /tmp/env.tar && \
+  mkdir /venv && \
+  cd /venv && tar xf /tmp/env.tar && \
+  rm /tmp/env.tar
 
-USER 1001
+# 'Un-pack the environment' - i.e. fix the path prefixes in the environment, so the environment is
+# equivalent to having been installed at /venv.
+RUN  /venv/bin/conda-unpack
```

Comment:
should we collapse all these `conda` commands into a single layer by chaining them with `&&`?

Reply:
That's a good question! I think it makes sense to collapse these layers (at least the conda packing and un-packing). 

I wonder if the the first `conda run` command might be cache-able on its own because it upgrades `pip` and almost all the images do that before installing context specific packages? I might be overthinking it.. I'll look into this